typed/remove_test.go

```diff
@@ -0,0 +1,139 @@
+package typed
+
+import (
+	"testing"
+
+	"sigs.k8s.io/structured-merge-diff/fieldpath"
+	"sigs.k8s.io/structured-merge-diff/schema"
+	"sigs.k8s.io/structured-merge-diff/value"
+)
+
+func TestRemoveShallowRemovesParents(t *testing.T) {
+
+	test, err := value.FromJSON([]byte(`
```

Comment:
Could use more test cases

Reply:
That test is really hard to parse and follow. I think you could do the same thing with a "deduced" schema (since you wouldn't have to specify the schema). There are also other schemas available throughout this test suite that you can use.

And yeah, it's good that we're testing how the two functions behave differently, but we need more examples :-/

Reply:
Right, I'll try deduced and add more cases.